tests/publiccloud/ssh_interactive_init.pm

```diff
@@ -0,0 +1,82 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: This tests will deploy the public cloud instance and prepare the ssh root connection
+#
+# Maintainer: Pavel Dostal <pdostal@suse.cz>
+
+use Mojo::Base 'publiccloud::basetest';
+use publiccloud::ssh_interactive;
+use testapi;
+use utils;
+
+sub run {
+    my ($self, $args) = @_;
+
+    # The user-console will be ocupated by the SSH tunnel
+    select_console 'user-console';
+    become_root;
+
+    # Create public cloud instance
+    my $provider = $self->provider_factory();
+    my $instance = $provider->create_instance(check_connectivity => 0);
+    $instance->check_ssh_port(timeout => 300);
+    $args->{my_provider} = $provider;
+    $args->{my_instance} = $instance;
+
+    # configure ssh client, fetch the instance ssh public key, do not use default $instance->ssh_opts
+    assert_script_run('curl ' . data_url('publiccloud/ssh_config') . ' -o ~/.ssh/config');
+    assert_script_run "ssh-keyscan " . $instance->public_ip . " > ~/.ssh/known_hosts";
+    $instance->ssh_opts("");
+
+    # configure ssh server, allow root and password login
+    $instance->run_ssh_command(cmd => "hostname -f");
+    $instance->run_ssh_command(cmd => "echo -e \"" . $testapi::password . "\\n" . $testapi::password . "\" | sudo passwd root");
+    $instance->run_ssh_command(cmd => "sudo sed -i \"s/PermitRootLogin no/PermitRootLogin yes/g\" /etc/ssh/sshd_config");
+    $instance->run_ssh_command(cmd => "sudo sed -i \"s/PasswordAuthentication no/PasswordAuthentication yes/g\" /etc/ssh/sshd_config");
+    $instance->run_ssh_command(cmd => "sudo cp /home/ec2-user/.ssh/authorized_keys /root/.ssh/");
+    $instance->run_ssh_command(cmd => "sudo systemctl reload sshd");
+
+    # Print ssh_serial_ready to serial console when interactive ssh session is initiated
+    $instance->run_ssh_command(cmd => "echo \"if tty -s; then echo ssh_serial_ready >> /dev/sshserial; fi\" > .bashrc; sudo cp .bashrc /root/");
+
+    # Save the instance IP to variable for publiccloud::ssh_interactive
+    set_var('PUBLIC_CLOUD_INSTANCE_IP', $instance->public_ip);
+
+    # Prepare the environment for the SSH tunnel
+    my $upload_port = get_required_var('QEMUPORT') + 1;
+    my $upload_host = testapi::host_ip();
+
+    $instance->run_ssh_command(
```

Comment:
Isn't this call, also something which should go into `publiccloud::ssh_interactive` ?